src/main.rs

```diff
@@ -144,24 +144,27 @@ async fn main() -> Result<(), Box<dyn Error>> {
 
     let offer_route = warp::post()
         .and(warp::body::json())
-        .map(move |offer: serde_json::Value| {
-            if let Some(offer_str) = offer.get("offer").and_then(|v| v.as_str()) {
-                // Convert the offer string to bytes and send
-                let offer_bytes = offer_str.as_bytes().to_vec();
-                let tx = offer_tx_clone.clone();
-                tokio::spawn(async move {
-                    if tx.send(offer_bytes).await.is_err() {
-                        eprintln!("Failed to send offer through the channel");
-                    }
-                });
-                warp::reply::with_status("Offer received", warp::http::StatusCode::OK)
-            } else {
-                warp::reply::with_status(
+        .map(
+            move |offer: serde_json::Value| match offer.get("offer").and_then(|v| v.as_str()) {
+                Some(offer_str) if offer_str.as_bytes().len() > 300 * 1024 / 8 => {
```

Comment:
It looks like this is checking for offers > 300/8 kb which is 37.5kb? I wasn't able to submit larger offers.

Reply:
removed the /8 and increased the message size in Gossipsub as well, because i was receiving [MessageTooLarge](https://docs.rs/libp2p-gossipsub/latest/libp2p_gossipsub/enum.PublishError.html#variant.MessageTooLarge) error.
Default size is 2048 bytes. [docs](https://docs.rs/libp2p-gossipsub/latest/libp2p_gossipsub/struct.ConfigBuilder.html#method.max_transmit_size)